### PR TITLE
[PO][FR] translation fixes

### PR DESCRIPTION
**Pull request by lourkeur**
## Description

I noticed a couple of mistranslations in my daily usage of fish, and another while editing the po file.

## TODO
- [x] run make po/fr.po
- [ ] build it and test it.
  I don't know how to get fish to print the "key with name X does not have any mapping" to print. The 3 others seem to work.
- [x] travis

I'd @ Xavier Douville but I can't find him on github.

### CODE REVIEW

**Comment by faho** [comment]
CC @PenegalECI.

File: po/fr.po [hunk 1]
```diff
@@ -204,7 +210,7 @@ msgstr "%ls: Aucune clef avec le nom '%ls' n'a été trouvée\n"
 #: src/builtin_bind.cpp:143
 #, fuzzy, c-format
 msgid "%ls: Key with name '%ls' does not have any mapping\n"
-msgstr "%ls: La fonction '%ls' n'existe pas\n"
+msgstr "%ls: La clef avec le nom '%ls' n'a aucune correspondance\n"
```
Comment (PenegalECI) [inline po/fr.po]: Attention : dans le reste du fichier, « clé » est préféré à « clef » ; autant uniformiser. Par ailleurs, vous pouvez retirer le _fuzzy_.

Beware: in the rest of the file, "clé" is preferred to "clef", so better use the preferred one. In addition, you can drop the _fuzzy_.

File: po/fr.po [hunk 2]
```diff
@@ -249,7 +255,7 @@ msgstr "%ls: '%ls' n'est pas un dossier\n"
 #: src/builtin_cd.cpp:55
 #, fuzzy, c-format
 msgid "%ls: The directory '%ls' does not exist\n"
-msgstr "%ls: La fonction '%ls' n'existe pas\n"
+msgstr "%ls: Le dossier '%ls' n'existe pas\n"
```
Comment (PenegalECI) [inline po/fr.po]: Drop the _fuzzy_.

File: po/fr.po [hunk 3]
```diff
@@ -264,7 +270,7 @@ msgstr "%ls: Option '%ls' inconnue\n"
 #: src/builtin_cd.cpp:76
 #, fuzzy, c-format
 msgid "%ls: Permission denied: '%ls'\n"
-msgstr "%ls: Nom de fonction illégal '%ls'\n"
+msgstr "%ls: Permission non accordée: '%ls'\n"
```
Comment (PenegalECI) [inline po/fr.po]: Drop the _fuzzy_.

File: po/fr.po [hunk 4]
```diff
@@ -1223,16 +1219,16 @@ msgstr "Une erreur est survenue lors du paramétrage du tube"
...
-#: src/exec.cpp:801
+#: src/exec.cpp:458
 #, c-format
 msgid "Unknown input redirection type %d"
 msgstr "Type de redirection d'entrée inconnu: %d"
 
+#: src/exec.cpp:827
+#, c-format
+msgid "Unknown function '%ls'"
+msgstr "Fonction inconnue '%ls'"
```
Comment (PenegalECI) [inline po/fr.po]: Utilisez « Fonction '%ls' inconnue », c’est du meilleur français et ça sonne moins _traduction littérale_.

Use "Fonction '%ls' inconnue", that's better French and sounds less _litteral translation_.

**Comment by PenegalECI** [comment]
@faho : I simply added comments, as I did not see I was reviewer; should I made them a review?

**Comment by faho** [comment]
>@faho : I simply added comments, as I did not see I was reviewer; should I made them a review?

@PenegalECI: I can't actually request a review from you as you're not a member of the organization. Of course the idea is that you're much more helpful than me in this context (I did take 4 years of french in school, but I was awful and I forgot most of it), so it'd be great if you reviewed it - even if it's not marked on github as such.

**Comment by faho** [comment]
Note: The test failure is unrelated - it seems Travis' macOS machines are overused at the moment (again), so some timeouts are failing.

**Comment by PenegalECI** [comment]
@faho : then the review is done, IMO.

File: po/fr.po [hunk 5]
```diff
@@ -76,11 +76,17 @@
 # to search                                               rechercher
 # array (RAID)                                            ensemble
 #
+#: /tmp/fish/implicit/share/completions/rustc.fish:3
```
[the remark this answers is not in the document]
Comment (lourkeur) [reply po/fr.po]: They were added automatically when I ran xgettext via the Makefile. They correspond to complete lines with no human readable message.

File: po/fr.po (hunk 1 above)
Comment (lourkeur) [reply po/fr.po]: will do

File: po/fr.po [hunk 6]
```diff
@@ -27376,129 +27426,134 @@ msgid "Force-push branch"
 msgstr "Pousser autoritairement la branche"
 
 #: /tmp/fish/implicit/share/completions/git.fish:226
+#, fuzzy
+msgid "Delete remote branch"
+msgstr "Supprimer la branche"
```
Comment (lourkeur) [inline po/fr.po]: la branche *distante*

the *remote* branch

File: po/fr.po [hunk 7]
```diff
@@ -44466,6 +44517,129 @@ msgstr "Ouvrir le profil spécifié dans un éditeur"
...
+msgstr ""
+
+#: /tmp/fish/implicit/share/completions/ngrok.fish:13
+msgid "log record format: 'term', 'logfmt', 'json'"
+msgstr ""
+
+#: /tmp/fish/implicit/share/completions/ngrok.fish:14
+#, fuzzy
+msgid "logging level"
+msgstr "Niveau de compression"
```
Comment (lourkeur) [inline po/fr.po]: incorrect

File: po/fr.po (hunk 6 above)
Comment (PenegalECI) [reply po/fr.po]: Indeed.

File: po/fr.po (hunk 7 above)
Comment (PenegalECI) [reply po/fr.po]: Welcome to my world. `make po/fr.po` tries to fuzzily complete empty translations by searching a translated close one, which makes stupid translations like that. Nevertheless, note they are fuzzy, i.e. marked as unreliable.

Comment (lourkeur) [reply po/fr.po]: I see. I'm fixing some of them now, particularly those that appear in my diffs, but not everything. And http://bitoduc.fr has been of no help today.

Your French is flawless btw

File: po/fr.po [hunk 8]
```diff
@@ -10065,269 +10062,274 @@ msgid "Optimize for the specified CPU architecture"
 msgstr ""
 
 #: /tmp/fish/implicit/share/completions/brew.fish:58
+#, fuzzy
+msgid "List installed top level formulae"
+msgstr "Limiter aux installés"
```
[the remark this answers is not in the document]
Comment (lourkeur) [reply po/fr.po]: Does "formules de premier niveau" sound good? I'm not a homebrew user

[the remark this answers is not in the document]
Comment (PenegalECI) [reply po/fr.po]: Neither do I, but it sounds good to me.

**Comment by faho** [comment]
Is the work here done?

**Comment by PenegalECI** [comment]
I just looked at the current version and it's go for merge, IMO.

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**